src/pyhf/modifiers/shapesys.py

```diff
@@ -66,16 +72,36 @@ def __init__(self, shapesys_mods, pdfconfig, mega_mods, batch_size=None):
             (len(shapesys_mods), self.batch_size or 1, 1),
         )
         # access field is shape (sys, batch, globalbin)
+
+        # reindex it based on current masking
+        self._reindex_access_field(pdfconfig)
+
+        self._precompute()
+        events.subscribe('tensorlib_changed')(self._precompute)
+
+    def _reindex_access_field(self, pdfconfig):
+        # access field maps to bin index, but needs to map to right index in
+        # the parameter set
```

Comment:
I know that this was approved when added in PR 519, but maybe the enumerate returns can be named more clearly to make it easier to follow. Though this would include also renaming the `s` and `t` perhaps to `syst_idx` and `batch_idx`. Though maybe I need a refresher on what the `self._access_field` is actually mapping.

Reply:
done